cmake/macros/macro_deal_ii_setup_target.cmake

```diff
@@ -103,6 +103,13 @@ MACRO(DEAL_II_SETUP_TARGET _target)
     COMPILE_DEFINITIONS "${DEAL_II_USER_DEFINITIONS};${DEAL_II_USER_DEFINITIONS_${_build}}"
     )
 
+  IF (DEAL_II_MSVC)
+    SET_PROPERTY(TARGET ${_target} PROPERTY
+      RUNTIME_OUTPUT_DIRECTORY_${_build} "${CMAKE_CURRENT_BINARY_DIR}"
+      )
+  ENDIF()
```

Comment:
This does not belong into ```DEAL_II_SETUP_TARGET``` and will break a number of valid use cases of this macro.
Can you please refactor this into ```DEAL_II_ADD_TEST```?